REALMC-12488: prevent errors being masked due to lack of captureStackTrace support

kpatel71716 opened the pull request:
I am fine not merging this PR, but wanted to put out the option if we/users would want visibility into errors while we properly implement Error.captureStackTrace()

This solution is a quick stopgap until [REALMC-10739](https://jira.mongodb.org/browse/REALMC-10739) is implemented. Instead of setting the stack property on the Error and returning a string representation, we just return an empty string. This allows users who are running into their errors being masked by captureStackTrace not being defined can proceed (e.g. the flaky sendgrid test is failing but we cannot see the actual error because we get back `captureStackTrace is not a function`).

The first commit allows us to write a Realm function that calls Error.captureStackTrace explicitly. The second commit is a stopgap to work with Error.captureStackTrace being called implicitly, like in a dependency. It's not great but it seems like both are needed until we figure out the root cause of we cannot recognize the Error [here](https://github.com/mongodb-forks/goja/blob/2d01d5628049e28e65fa8d818b44fb7a25858b25/vm.go#L1813). Unfortunately that would mean that any time the captureStackTrace prop is accessed, we would return that function even if though it is not being called on an Error.

Without this, it's hard to tell why the sendgrid test is actually failing inconsistently

Gabri3l reviewed (approved):
I'm actually fine with this for the time being, as you mentioned it'll help us let some dependencies surface errors that are otherwise masked by missing `captureStackTrace`. You have a mem test failing so make sure to address that before merging.

kpatel71716 commented:
Updated the existing test  in memory_test.go and added a TODO to add another test once we fully implement Error.captureStackTrace. If we change the default behavior to always set stack as "", the updated test should correctly fail

Gabri3l commented:
I separated the work to stub `capureStackTrace` since this does not address the flakiness yet and we already had tickets in place for this. Now with the new ticket we address "stubbing" and with the other linked ticket we will actually implement it.
